### ISSUE TITLE
Vue 3 package dependency compatibility

### ISSUE THREAD

**Issue by dxps** [tserkov/vue-sse#38]
Hello and thanks for building this package!

I'd like to use it in a Vue 3 based project, but the default installation fails due to package dependency incompatibility:

```shell
$ npm install --save vue-sse
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: txn_ui@0.1.0
npm ERR! Found: vue@3.2.31
npm ERR! node_modules/vue
npm ERR!   vue@"^3.2.13" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vue@"^2.0.0" from vue-sse@2.5.0
npm ERR! node_modules/vue-sse
npm ERR!   vue-sse@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

```

I saw that [the docs](https://github.com/tserkov/vue-sse#vue-3) are saying that is compatible with Vue 3, and the [vue-sse-example](https://github.com/tserkov/vue-sse-example) is based on Vue 2.

Shall I `--force` install it?

Thanks!

**Comment by dxps** [comment]
Meanwhile, I used that (only?) option, and it's all working fine (at least a minimally functional usage).
I'll leave this issue open, so anyone can provide any feedback about it. Otherwise, it could be closed.

**Comment by tserkov** [comment]
Thanking you for bringing this to my attention! I admittedly haven't used vue-sse with any of my Vue 3 projects yet.

The plugin [detects Vue 2 vs Vue 3 on install](https://github.com/tserkov/vue-sse/blob/v2/src/sse-manager.js#L4-L11), so there's absolutely no danger is using `--force`, as you've discovered, to override the peer dependency.

What I should do is modify [`peerDependencies`](https://github.com/tserkov/vue-sse/blob/v2/package.json#L57-L59) to `"^2.0.0 || ^3.0.0"` so npm doesn't panic. Fix inbound this week!

**Comment by dxps** [comment]
Lovely, thanks James! :blush:

**Comment by sfishmonger** [comment]
Hello, first of all, thanks for this package
Do we have any update about this topic ?

**Comment by dxps** [comment]
@sfishmonger  In my case, using this package was a complete success! :partying_face: 
But sure, @tserkov could do (or any of us actually send a PR) the aforementioned update so that no --force is needed.

**Comment by sfishmonger** [comment]
Hello, 
yes, it's work fine with the '-- force' option, but I have to use this option on every projet including vue-sse directly or projets that use module with dependencies with vue-see.
It's a nice workarround indeed, but not as clean as I want 😄

**Comment by tserkov** [comment]
Sorry for the delay! I should have left this issue as Open. 😅

Fixed & published as v2.5.1. Thanks all!

**Comment by sfishmonger** [comment]
thanks :)
